src/map/lua/lua_baseentity.cpp

```diff
@@ -366,6 +366,106 @@ inline int32 CLuaBaseEntity::isJugPet(lua_State* L)
 
 //======================================================//
 
+inline int32 CLuaBaseEntity::getCharmChance(lua_State* L)
```

Comment:
Refactored so that the charm and lua functions are now using the new GetCharmChance function to reduce the amount of duplicate/similar code.